### PR TITLE
Rename kibana substrings with OpenSearchDashboards in class name, method name and comments

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__
Rename `kibana` substrings with `OpenSearchDashboards`  in class name, method name and comments

 4. __Why these changes are required?__ 
 OpenSearch migration

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
No changes in behavior

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 UT
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: securityconfig/roles.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ _meta:
   type: "roles"
   config_version: 2
 
-# Restrict users so they can only view visualization and dashboard on kibana
+# Restrict users so they can only view visualization and dashboard on Kibana
```
Comment (andy840314) [inline securityconfig/roles.yml:5]: Should this be renamed to OpenSearchDashboard?

File: securityconfig/roles.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ _meta:
...
 kibana_read_only:
```
Comment (andy840314) [inline securityconfig/roles.yml:6]: Do we need to rename this?

**Review by vrozov — DISMISSED** [review]
The change makes code less readable compared to original

**Comment by cliu123** [comment]
> The change makes code less readable compared to original

@vrozov Could you please elaborate a little more? What is less readable? prefix? 
Suggestions to improve? removing the prefix?

**Comment by vrozov** [comment]
@cliu123 Code becomes less readable. For example naming variable that gets "kibana_url" settings as `openSearchDashboardsRootUrl` makes code less readable.

File: securityconfig/roles.yml (hunk 2 above)
Comment (cliu123) [reply securityconfig/roles.yml:6]: That's not in the scope of this PR.

File: securityconfig/roles.yml (hunk 1 above)
Comment (cliu123) [reply securityconfig/roles.yml]: Done

File: src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java [hunk 3]
```diff
@@ -95,7 +95,7 @@ public HTTPSamlAuthenticator(final Settings settings, final Path configPath) {
 
             rolesKey = settings.get("roles_key");
             subjectKey = settings.get("subject_key");
-            kibanaRootUrl = settings.get("kibana_url");
+            openSearchDashboardsRootUrl = settings.get("kibana_url");
```
[the remark this answers is not in the document]
Comment (cliu123) [reply src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java]: Done

File: src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenDistroSecuritySSLCertsDashboardsInfoActionTests.java [hunk 4]
```diff
@@ -30,7 +30,7 @@
 import java.util.List;
 import java.util.Map;
 
-public class OpenDistroSecuritySSLCertsInfoActionTests extends SingleClusterTest {
+public class OpenDistroSecuritySSLCertsDashboardsInfoActionTests extends SingleClusterTest {
```
Comment (andy840314) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenDistroSecuritySSLCertsDashboardsInfoActionTests.java:33]: Why is this class renamed?

File: src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/TenantDashboardsInfoActionTest.java [hunk 5]
```diff
@@ -28,7 +28,7 @@
 import org.mockito.invocation.InvocationOnMock;
 import org.mockito.stubbing.Answer;
 
-public class TenantInfoActionTest extends AbstractRestApiUnitTest {
+public class TenantDashboardsInfoActionTest extends AbstractRestApiUnitTest {
```
[the remark this answers is not in the document]
Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/TenantDashboardsInfoActionTest.java]: Reverted back

File: src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenDistroSecuritySSLCertsDashboardsInfoActionTests.java (hunk 4 above)
Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenDistroSecuritySSLCertsDashboardsInfoActionTests.java]: Reverted back.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java [hunk 6]
```diff
@@ -119,14 +119,14 @@ public ReplaceResult replaceKibanaIndex(final ActionRequest request, final Strin
         }
 
         //intercept when requests are not made by the kibana server and if the kibana index/alias (.kibana) is the only index/alias involved
```
Comment (andy840314) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java:121]: Not sure if we need to rename kibana to openSearchDashboards in these comments.
All the comments that contain kibana: 121, 145, 155, 230

File: src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v6/ConfigV6Test.java [hunk 7]
```diff
@@ -79,7 +79,7 @@ public ConfigV6Test(boolean omitDefaults) {
     }
 
     @Test
-    public void testKibana() throws Exception {
+    public void testDashboards() throws Exception {
         ConfigV6.Kibana kibana;
```
Comment (andy840314) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v6/ConfigV6Test.java:83]: Should the class `ConfigV6.Kibana` be renamed to `ConfigV6.OpenSearchDashboards`? Also the variable name should be renamed.

File: src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v7/ConfigV7Test.java [hunk 8]
```diff
@@ -72,7 +72,7 @@ public ConfigV7Test(boolean omitDefaults) {
     }
 
     @Test
-    public void testKibana() throws Exception {
+    public void testDashboards() throws Exception {
         ConfigV7.Kibana kibana;
```
Comment (andy840314) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v7/ConfigV7Test.java:76]: Should the class `ConfigV7.Kibana` be renamed to `ConfigV7.OpenSearchDashboards`? Also the variable name should be renamed.

**Comment by andy840314** [comment]
I found that there are some `kibana` keywords in loggers, should we rename that in this PR?
e.g. `PrivilegesInterceptorImpl.java`  line 194, 206

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java (hunk 6 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java:121]: These are indices, which is not in the scope of this PR. We'll change these things when we migrate indices.

**Comment by cliu123** [comment]
> I found that there are some `kibana` keywords in loggers, should we rename that in this PR?
> e.g. `PrivilegesInterceptorImpl.java` line 194, 206

The plan for these changes will be discussed for indices migration.

File: src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v6/ConfigV6Test.java (hunk 7 above)
Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v6/ConfigV6Test.java:83]: As we discussed, we'll take care of this change in the reset of the migration tasks.

File: src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v7/ConfigV7Test.java (hunk 8 above)
Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/v7/ConfigV7Test.java:76]: As we discussed, we'll take care of this change in the reset of the migration tasks.

**Review by andy840314 — DISMISSED** [review]

File: src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java [hunk 9]
```diff
@@ -145,7 +145,7 @@ private void initMisc(HashMap<String, Object> configProperties) {
 
     private void initSpEndpoints(HashMap<String, Object> configProperties) {
         configProperties.put(SettingsBuilder.SP_ASSERTION_CONSUMER_SERVICE_URL_PROPERTY_KEY,
-                this.buildKibanaAssertionConsumerEndpoint(this.opensearchSettings.get("kibana_url")));
+                this.buildDashboardsConsumerEndpoint(this.opensearchSettings.get("kibana_url")));
```
Comment (debjanibnrj) [inline src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java:148]: I would suggest renaming this to buildAssertionConsumerEndpoint. The assertion consumer service (ACS) endpoint is a location to which the SSO tokens are sent. Generic name would be good here.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/rest/DashboardsInfoAction.java [hunk 10]
```diff
@@ -94,8 +94,8 @@ public void accept(RestChannel channel) throws Exception {
                     builder.field("user_name", user==null?null:user.getName());
                     builder.field("not_fail_on_forbidden_enabled", evaluator.notFailOnForbiddenEnabled());
                     builder.field("kibana_mt_enabled", evaluator.multitenancyEnabled());
```
Comment (debjanibnrj) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/rest/DashboardsInfoAction.java:96]: It appears that these field names have been changed in the security-dashboards-plugin - https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/backend/opendistro_security_plugin.ts#L40-L42

Can we make sure field names are consistent?

**Review by dblock — APPROVED** [review]
+1 on the comments that suggest generalizing names where possible, can be done after this is merged

File: src/main/java/com/amazon/opendistroforelasticsearch/security/rest/DashboardsInfoAction.java (hunk 10 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/rest/DashboardsInfoAction.java:96]: This is in the scope of another migration task for Kibana that @hsiang9431-amzn is going to take care of.

File: src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java (hunk 9 above)
Comment (cliu123) [reply src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java]: Changed.

**Review by andy840314 — APPROVED** [review]

**Review by debjanibnrj — APPROVED** [review]

**Comment by cliu123** [comment]
> @cliu123 Code becomes less readable. For example naming variable that gets "kibana_url" settings as `openSearchDashboardsRootUrl` makes code less readable.

Changed it back. It'll be renamed in one of the PRs following.

**Pull request merged**